hedera-node/src/main/java/com/hedera/services/grpc/marshalling/FractionalFeeAssessor.java

```diff
@@ -62,18 +65,29 @@ public ResponseCodeEnum assessAllFractional(
 				return CUSTOM_FEE_OUTSIDE_NUMERIC_RANGE;
 			}
 
-			unitsLeft -= assessedAmount;
-			if (unitsLeft < 0) {
-				return INSUFFICIENT_PAYER_BALANCE_FOR_CUSTOM_FEE;
+			boolean chargeSender = fee.getFractionalFeeSpec().getNetOfTransfers();
+
+			if(!chargeSender) { // charge receiver
+				unitsLeft -= assessedAmount;
+				if (unitsLeft < 0) {
+					return INSUFFICIENT_PAYER_BALANCE_FOR_CUSTOM_FEE;
+				}
+				final var creditsToReclaimFrom = changeManager.creditsInCurrentLevel(denom);
+				try {
+					reclaim(assessedAmount, creditsToReclaimFrom);
+				} catch (ArithmeticException ignore) {
+					return CUSTOM_FEE_OUTSIDE_NUMERIC_RANGE;
+				}
 			}
-			final var creditsToReclaimFrom = changeManager.creditsInCurrentLevel(denom);
-			try {
-				reclaim(assessedAmount, creditsToReclaimFrom);
-			} catch (ArithmeticException ignore) {
-				return CUSTOM_FEE_OUTSIDE_NUMERIC_RANGE;
+			else {
+				final long totalCharge = initialUnits + assessedAmount;
```

Comment:
We don't want to involve the `HederaLedger` here I think...it will be much simpler to inject the `FixedFeeAssessor` into the `FractionalFeeAssessor` and use it to assess the added fee (as the `RoyaltyFeeAssessor` does [here](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/grpc/marshalling/RoyaltyFeeAssessor.java#L80)).